python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -289,30 +288,22 @@ def predict(self, X, output_class=True, algo='auto', threshold=0.5,
         y : Dask cuDF dataframe or CuPy backed Dask Array (n_rows, 1)
 
         """
-        if self.num_classes > 2 or predict_model == "CPU":
+        if predict_model == "CPU" or self.num_classes > 2:
             preds = self.predict_model_on_cpu(X,
                                               convert_dtype=convert_dtype)
 
         else:
             preds = \
-                self.predict_using_fil(X, output_class=output_class,
-                                       algo=algo,
-                                       threshold=threshold,
-                                       num_classes=self.num_classes,
-                                       convert_dtype=convert_dtype,
-                                       predict_model="GPU",
-                                       fil_sparse_format=fil_sparse_format,
-                                       delayed=delayed)
+                self._predict_using_fil(X, output_class=output_class,
+                                        algo=algo,
+                                        threshold=threshold,
+                                        convert_dtype=convert_dtype,
+                                        predict_model="GPU",
```

Comment:
It's interesting that we have to pass predict_model=GPU to predict_using_fil... is this necessary? Seems like FIL only runs on gpu so it was surprising.

Reply:
Its not necessary to pass `predict_model="GPU"` since the single gpu implementation of predict by default uses `predict_model="GPU"`.